README.md

```diff
@@ -16,85 +16,96 @@ Without needing to worry about instruction selection, register pressure, and so
 The ultimate goal of Slinky is to make automatic scheduling of pipelines reliable and fast enough to implement a just-in-time optimization engine at runtime.
 
 ## Graph description
-Pipelines are described by operators called `func`s and connected by `buffer_expr`s.
-`func` has a list of `input` and `output` objects.
-A `func` can have multiple `output`s, but all outputs must be indexed by one set of dimensions for the `func`.
-An `input` or `output` is associated with a `buffer_expr`.
-An `output` has a list of dimensions, which identify variables (`var`) used to index the corresponding dimension of the `buffer_expr`.
-An `input` has a list of bounds expressions, expressed as an inclusive interval `[min, max]`, where the bounds can depend on the variables from the output dimensions.
+A pipeline is a directed acyclic graph, where the nodes are `buffer_expr`s, and the edges are `func`s.
+A `func` is said to "consume" its input buffer(s), and "produce" its output buffers, of which there can be more than one.
+A node that is not produced by any `func` is a pipeline input, and a node that is not consumed by a `func` is a pipeline output.
 
-The actual implementation of a `func` is a callback taking a single argument `eval_context`.
-This object contains the state of the program at the time of the call.
-Values of any symbol currently in scope at the time of the call can be accessed in the `eval_context`. 
+A function encapsulates:
+
+- a symbolic description of data dependecies
```

Comment:
I think this needs more elaboration somewhere. The old description described this in more detail.